corehq/apps/api/validation.py

```diff
@@ -100,6 +104,21 @@ def validate_custom_data(self, custom_data, profile_name):
         spec = {'data': custom_data, 'user_profile': profile_name}
         return custom_data_validator.validate_spec(spec)
 
+    def validate_custom_data_with_profile(self, custom_data, profile_name):
+        if custom_data is None or profile_name is None:
+            return
+
+        errors = []
+        profile = self.profiles_by_name.get(profile_name)
+
+        system_fields = set(profile.fields.keys()) if profile else set()
+        system_fields.add(PROFILE_SLUG)
+
+        for key in custom_data.keys():
+            if key in system_fields:
+                errors.append(_("'{}' cannot be set directly").format(key))
```

Comment:
Not blocking, but would it be helpful if this message was a little more descriptive or instructive? By that I mean pointing out that this field is part of a profile and the individual fields can't be updated and the user would need to update the profile.

Reply:
[09d6a55](https://github.com/dimagi/commcare-hq/pull/35575/commits/09d6a55393934dbf26b6bc728ca720dac7c99725)